### DISCUSSION TITLE
For 2.6 Is it worth having new default settings for TripleA.vmoptions ?

### DISCUSSION THREAD

**Discussion by TheDog-GH** [triplea-game/triplea#10610] [General]
@asvitkine @Cernel @DanVanAtta @RoiEXLab and others.

Im on Windows 10 64bit, with 8GB and my hardware appears under used with default settings of;
-Xmx2048M
-Xms2048M
-Xss1250K

Cernel has changed his values to; (Anecdotally, he also says it improves game play)
-Xmx4096M
-Xms4096M

I have changed my TripleA.vmoptions to the following;
-Xmx3072M
-Xms3072M
-Xss1536K

Anecdotally increasing the values in TripleA.vmoptions appears to make it run faster by not using the hard disk as much and using more RAM.  These maps seem to benefit from the increases;
270BC Wars
Warcraft
The Shogun
War of the Relics
Twelve Clans
Greyhawk Wars
Another World
Im sure there are others but I have not played them.

Perhaps a custom TripleA.vmoptions file could be written, detecting the installed RAM ?
Thoughts?

**Comment by asvitkine** [comment]
Install4j apparently has an option to set a dynamic heap size based on system RAM:

https://www.ej-technologies.com/resources/install4j/help/doc/help.pdf

```
To set an -Xmx value that depends on the total memory of the target system, you can use a "Set
a variable action" to calculate the numeric part of the -Xmx value using the utility method
SystemInfo.getPhysicalMemory() and use that variable in the "VM options" property of the
"Add VM options"action. For example, in order to use 50% of the total memory for the maximum
heap size, you do the following after the "Install files" action: ...
```

Maybe we could use that.

**Comment by TheDog-GH** [comment]
Here is some background, but basically saying the same, TripleA.vmoptions or memory allocation needs to change.

https://github.com/triplea-game/triplea/issues/9654
https://github.com/triplea-game/triplea/issues/10440

https://forums.triplea-game.org/topic/3076/java-faster-triplea-on-windows-10
https://forums.triplea-game.org/topic/3133/get-ready-for-our-new-game-the-shogun-coming-soon

**Comment by DanVanAtta** [comment]
For java memory management, the rule of thumb is "don't roll your own unless you really know what you are doing".

This setting for example means any TripleA instance will consume 4GB of memory. `-Xms4096M`
If you have one game open and the lobby, you will then consume nearly 8GB of memory, opening another game would put you at 12GB memory usage and so on.

The JVM defaults are generally very smart and are dynamic based upon the available memory. Overriding these is not necessarily to be done lightly. It is potentially more a problem that we specify the 'vmoption' defaults at all rather than allowing the JVM to use default (and very often optimal) algorithms for memory management.

**Comment by DanVanAtta** [comment]
In short, I should have mentioned - we may be best off deleting the 'vmoptions' file entirely. The default values chosen may very well be larger/better than what we are specifying.
